### PR TITLE
Remove V1 Primitive Support

### LINKED ISSUE #1857: Remove V1 support
**What is the expected feature or enhancement?**

Qiskit Runtime is dropping V1 primitives support on August 15. This repo should be updated to also remove V1 support. 

**Acceptance criteria**

- V1 related code should be cleaned up
- `from qiskit_ibm_runtime import Estimator` should import V2 instead

### PR DESCRIPTION
**Pull request by kt474**
### Details and comments
Fixes #1857

### CODE REVIEW

File: qiskit_ibm_runtime/__init__.py [hunk 1]
```diff
@@ -218,14 +216,9 @@
...
-    EstimatorV1 as Estimator,
-)
-from .sampler import (  # pylint: disable=reimported
-    SamplerV2,
-    SamplerV1,
-    SamplerV1 as Sampler,
+    EstimatorV2 as Estimator,
 )
+from .sampler import SamplerV2, SamplerV2 as Sampler  # pylint: disable=reimported
 from .options import Options, EstimatorOptions, SamplerOptions, OptionsV2
```
Comment (jyu00) [inline qiskit_ibm_runtime/__init__.py:222]: There are a number of options class, such as `Options` and `TranspilationOptions` that are V1 only.

File: qiskit_ibm_runtime/__init__.py [hunk 2]
```diff
@@ -218,15 +214,15 @@
...
     EstimatorV2,
-    EstimatorV1,
-    EstimatorV1 as Estimator,
+    EstimatorV2 as Estimator,
 )
-from .sampler import (  # pylint: disable=reimported
-    SamplerV2,
-    SamplerV1,
-    SamplerV1 as Sampler,
+from .sampler import SamplerV2, SamplerV2 as Sampler  # pylint: disable=reimported
```
Comment (ptristan3) [inline qiskit_ibm_runtime/__init__.py:219]: repeated SamplerV2

File: qiskit_ibm_runtime/base_primitive.py [hunk 3]
```diff
@@ -24,15 +23,10 @@
...
-from qiskit.providers.options import Options as TerraOptions
 from qiskit.providers.backend import BackendV1, BackendV2
 
-from .provider_session import get_cm_session as get_cm_provider_session
-
-from .options import Options
 from .options.options import BaseOptions, OptionsV2
-from .options.utils import merge_options, set_default_error_levels, merge_options_v2
-from .runtime_job import RuntimeJob
+from .options.utils import merge_options, merge_options_v2
```
Comment (ptristan3) [inline qiskit_ibm_runtime/base_primitive.py:29]: `merge_options` function still makes sense?

File: qiskit_ibm_runtime/utils/estimator_result_decoder.py [hunk 4]
```diff
@@ -27,13 +25,8 @@ class EstimatorResultDecoder(ResultDecoder):
     @classmethod
     def decode(  # type: ignore # pylint: disable=arguments-differ
         cls, raw_result: str
-    ) -> Union[EstimatorResult, PrimitiveResult]:
+    ) -> Union[PrimitiveResult]:
```
Comment (ptristan3) [inline qiskit_ibm_runtime/utils/estimator_result_decoder.py:28]: `Union` is not needed

File: qiskit_ibm_runtime/utils/sampler_result_decoder.py [hunk 5]
```diff
@@ -26,29 +23,10 @@ class SamplerResultDecoder(ResultDecoder):
     """Class used to decode sampler results."""
 
     @classmethod
-    def decode(cls, raw_result: str) -> Union[SamplerResult, PrimitiveResult]:
+    def decode(cls, raw_result: str) -> Union[PrimitiveResult]:
```
Comment (ptristan3) [inline qiskit_ibm_runtime/utils/sampler_result_decoder.py:26]: also here `Union` is not needed

File: qiskit_ibm_runtime/base_primitive.py (hunk 3 above)
Comment (kt474) [reply qiskit_ibm_runtime/base_primitive.py:29]: I think we still need it for those that want to pass options as a `dict`

**Review by ptristan3 — APPROVED** [review]

**Pull request merged**